### PR TITLE
chore: binary build cleanup

### LINKED ISSUE #25077: Remove Dev Dep Patches from Binary
### What would you like?

Currently we are shipping patchs for dev-dependencies in the binary code. These aren't needed for our users. This issue is to clean that up.

Screen Shot 2022-12-08 at 3 22 27 PM

### PR DESCRIPTION
**Pull request by emilyrohrbough**
Closes https://github.com/cypress-io/cypress/issues/25077

**1. removed the dev dependency patches from the binary. Reduced from 40 to 16 shipped patches**
| [before](https://app.circleci.com/pipelines/github/cypress-io/cypress/52067/workflows/4a14f32f-1d78-49fd-b159-ecc3496cccff/jobs/2148884) | [after](https://app.circleci.com/pipelines/github/cypress-io/cypress/52067/workflows/4a14f32f-1d78-49fd-b159-ecc3496cccff/jobs/2148884)
--| --
Screen Shot 2023-05-01 at 9 29 31 AM | Screen Shot 2023-04-28 at 5 07 03 PM

**2. bumps yarn concurrency from 1 to 4 for faster build times. quick check shows 1 min + improvement depending on machine platform**
NOTE: reference x-create-build-artifacts job
|Platform | Average (last 30 days Mar 29-April 288) | Times on this Branch |
| --|--|--|
| linux-arm64 | Screen Shot 2023-04-28 at 4 59 27 PM | [Job Time 9m 19s](https://app.circleci.com/pipelines/github/cypress-io/cypress/52060/workflows/30dd781b-be8e-4a98-9c81-49f476abab9c/jobs/2148332) |
| linux-x64 | Screen Shot 2023-04-28 at 4 59 56 PM | [Job Time 12 m 5s](https://app.circleci.com/pipelines/github/cypress-io/cypress/52057/workflows/0c9fb9bc-bf5d-4b5e-a397-66637bc547b2/jobs/2148163) |
| darwin-arm64 | Screen Shot 2023-04-28 at 4 59 12 PM | [Job Time 20 m 9 s](https://app.circleci.com/pipelines/github/cypress-io/cypress/52060/workflows/0ac7d8f0-2845-428a-9a46-29a2fb853127/jobs/2148352) |
| darwin-x64 | Screen Shot 2023-04-28 at 4 58 58 PM | [Job Time 27 m 39 s](https://app.circleci.com/pipelines/github/cypress-io/cypress/52060/workflows/0ce96300-a373-4939-a4df-a77b82e6d8b4/jobs/2148359) |
| window | Screen Shot 2023-04-28 at 5 00 21 PM | [Job Time 24 m 35 s](https://app.circleci.com/pipelines/github/cypress-io/cypress/52060/workflows/0c480db3-0fee-4212-a0a9-94e0676a224e/jobs/2148365) |

### PR Tasks

- [n/a] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Comment by ryanthemanuel** [comment]
How much did this affect the binary size out of curiosity?

File: tooling/v8-snapshot/cache/darwin/snapshot-meta.json [hunk 1]
```diff
@@ -960,7 +960,7 @@
     "./packages/socket/node_modules/socket.io/node_modules/ws/lib/websocket.js",
     "./packages/ts/register.js",
     "./packages/types/index.js",
-    "./tooling/v8-snapshot/dist/setup/v8-snapshot-entry-cy-in-cy.js"
+    "./tooling/v8-snapshot/dist/setup/v8-snapshot-entry.js"
```
Comment (ryanthemanuel) [inline tooling/v8-snapshot/cache/darwin/snapshot-meta.json]: Let's revert these changes. This is only valid really for the binary build.

**Comment by emilyrohrbough** [comment]
@ryanthemanuel Great question -- looks like minimal change in the binary size. I updated the before/after binary size impact in the summary - when comparing the runs - they were to different operating systems, which is why it looked like there should have been more.

path |  [before](https://app.circleci.com/pipelines/github/cypress-io/cypress/52067/workflows/4a14f32f-1d78-49fd-b159-ecc3496cccff/jobs/2148884) | [after](https://app.circleci.com/pipelines/github/cypress-io/cypress/52067/workflows/4a14f32f-1d78-49fd-b159-ecc3496cccff/jobs/2148884)
--|--| --
 | | Screen Shot 2023-05-01 at 9 31 48 AM|Screen Shot 2023-05-01 at 9 31 58 AM
/tmp/cypress-build/linux/build/Cypress/browser_v8_context_snapshot.bin | 116M | 115M
/tmp/cypress-build/linux/build/Cypress/resources | 210M | 209M
zip size | 169 MB | 169 MB

File: tooling/v8-snapshot/cache/darwin/snapshot-meta.json (hunk 1 above)
Comment (emilyrohrbough) [reply tooling/v8-snapshot/cache/darwin/snapshot-meta.json]:
```suggestion
    "./tooling/v8-snapshot/dist/setup/v8-snapshot-entry-cy-in-cy.js"
```

File: tooling/v8-snapshot/cache/darwin/snapshot-meta.json [hunk 2]
```diff
@@ -4235,8 +4235,6 @@
     "./packages/extension/index.js",
     "./packages/extension/lib/extension.js",
     "./packages/extension/lib/util.js",
-    "./packages/frontend-shared/cypress/e2e/prod-dependencies.ts",
-    "./packages/frontend-shared/cypress/e2e/v8-snapshot-entry.ts",
     "./packages/graphql/node_modules/chalk/source/templates.js",
```
Comment (emilyrohrbough) [inline tooling/v8-snapshot/cache/darwin/snapshot-meta.json]:
```suggestion
    "./packages/frontend-shared/cypress/e2e/prod-dependencies.ts",
    "./packages/frontend-shared/cypress/e2e/v8-snapshot-entry.ts",
    "./packages/graphql/node_modules/chalk/source/templates.js",
```

**Review by ryanthemanuel — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]
